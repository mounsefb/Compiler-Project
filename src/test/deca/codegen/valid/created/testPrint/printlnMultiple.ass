	TSTO #7
	BOV stack_overflow_error
	ADDSP #20
; start main program
; Main program
; Beginning of main instructions:
	LOAD #5, R2
	OPP R2, R2
	OPP R2, R2
	OPP R2, R2
	OPP R2, R2
	OPP R2, R2
	FLOAT R2, R2
	STORE R2, 1(GB)
	LOAD #6, R2
	OPP R2, R2
	OPP R2, R2
	OPP R2, R2
	STORE R2, 2(GB)
	WSTR "Hello"
	WSTR "world"
	WSTR "!"
	LOAD #3, R1
	WINT
	LOAD #8, R2
	OPP R2, R2
	LOAD R2, R1
	WINT
	WNL
	LOAD #4, R3
	OPP R3, R3
	LOAD R3, R1
	WINT
	WNL
	LOAD 1(GB), R1
	WFLOAT
	WNL
	LOAD 1(GB), R4
	OPP R4, R4
	LOAD R4, R1
	WFLOAT
	WNL
	LOAD 2(GB), R1
	WINT
	WNL
	LOAD 2(GB), R5
	OPP R5, R5
	LOAD R5, R1
	WINT
	WNL
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
; code des m√©thodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
