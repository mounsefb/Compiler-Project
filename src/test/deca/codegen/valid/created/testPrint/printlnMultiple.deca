// Description :
//    Test de l'affichage de plusieurs chaînes de caractères avec println
//
// Résultats :
//     Helloworld!3-8
//     -4
//     -5.00000e+00
//     5.00000e+00
//     -6
//     6


{
   float x = -----5;
   int y = ---6;
   println("Hello", "world", "!", 3, -8);
   println(-4);
   println(x);
   println(-x);
   println(y);
   println(-y);
}