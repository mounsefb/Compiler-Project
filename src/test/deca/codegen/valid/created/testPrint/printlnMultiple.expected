Helloworld!3-8
-4
-5.00000e+00
5.00000e+00
-6
6
